src/apps/backend/modules/config/internals/config_file.py

```diff
@@ -0,0 +1,127 @@
+from __future__ import annotations
+
+import os
+from pathlib import Path
+from typing import Any, Optional
+
+import yaml
+
+
+class ConfigFile:
+    """
+    Represents a configuration file that can be loaded, merged, and updated with environment variables.
+    """
+
+    def __init__(self, filename: str, config_path: Path):
+        self.filename = filename
+        self.config_path = config_path
+        self.content: dict[str, Any] = {}
+
+    def load(self) -> None:
+        """
+        Loads the content of the configuration file.
+        """
+        file_path = self.config_path / self.filename
+        try:
+            with open(file_path, "r", encoding="utf-8") as file:
+                self.content = yaml.safe_load(file) or {}
+        except FileNotFoundError:
+            from modules.logger.logger import Logger
+
+            Logger.error(message=f"Config file '{self.filename}' not found.")
+
+    def merge(self, other: ConfigFile) -> None:
+        """
+        Merges the content of another ConfigFile into this one.
+        """
+        self.content = self._deep_merge(self.content, other.content)
+
+    def replace_with_env_variables(self) -> None:
+        """
+        Replaces placeholders in the configuration file with corresponding environment variables.
+        """
+        self.content = self._replace_with_env_values(self.content)
+
+    def get_content(self) -> dict[str, Any]:
+        """
+        Returns the loaded content of the config file.
+        """
+        return self.content
+
+    @staticmethod
+    def _deep_merge(dict1: dict[str, Any], dict2: dict[str, Any]) -> dict[str, Any]:
+        """
+        Deep merge two dictionaries.
+        """
+        for key, value in dict2.items():
+            if isinstance(value, dict) and key in dict1 and isinstance(dict1[key], dict):
+                dict1[key] = ConfigFile._deep_merge(dict1[key], value)
+            else:
+                dict1[key] = value
+        return dict1
+
+    @staticmethod
+    def _replace_with_env_values(data: dict[str, Any]) -> dict[str, Any]:
```

Comment:
It is not clear why a config file has or knows about "env" variable. Something about the abstraction does not looks right.